lib/ts/string_view.h

```diff
@@ -1220,7 +1220,13 @@ template <class _Type, class _Traits> struct hash<ts::basic_string_view<_Type, _
   size_t
   operator()(string_type const &x) const
   {
-    return hash<typename string_type::const_pointer>()(x.data());
+// not what I would normally do.. but better than making a custom hash function at the moment.
+// This should also mean we have some consistent behavior with std code
+#if defined(__linux__)
```

Comment:
I think this is not OS dependent but compiler dependent. E.g. `__gnuc` and `__clang`.

Reply:
Oddly, I tried this PR using clang on fedora 27 and it builds fine. Not a real concern, then.